src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,9 +37,24 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        splittingAddressIntoFourDifferentFormat(trimmedAddress);
         this.value = trimmedAddress;
     }
 
+
+    public void splittingAddressIntoFourDifferentFormat (String trimmedAddress) throws IllegalValueException {
```

Comment:
Don't forget the header comment. I believe the method name should be in simple present tense, e.g. `splitAddress...`.